android/src/main/java/com/tailscale/ipn/ui/view/PingView.kt

```diff
@@ -0,0 +1,211 @@
+                                  thickness = 3.dp,
+                                  backgroundShader =
+                                      DynamicShader.verticalGradient(
+                                          arrayOf(
+                                              ts_color_light_blue.copy(alpha = 0.4f),
+                                              ts_color_light_blue.copy(alpha = 0f))),
+                              ),
+                          )),
+                      startAxis =
+                          rememberStartAxis(
+                              valueFormatter =
+                                  CartesianValueFormatter.decimal(
+                                      decimalFormat = DecimalFormat("#;−#")),
+                              itemPlacer =
+                                  remember { AxisItemPlacer.Vertical.count(count = { 4 }) },
+                              guideline =
+                                  rememberLineComponent(
+                                      color = MaterialTheme.colorScheme.outlineVariant,
+                                      shape =
+                                          remember {
+                                            Shape.dashed(
+                                                dashLength = 18.dp,
+                                                gapLength = 24.dp,
+                                                shape = Shape.Rectangle,
+                                            )
+                                          }),
+                              label =
+                                  rememberAxisLabelComponent(
+                                      color = MaterialTheme.colorScheme.onBackground,
+                                      padding = Dimensions.of(horizontal = 6.dp)))),
+              model = CartesianChartModel(LineCartesianLayerModel.build { series(pingValues) }),
+          )
+        } else {
+          errorMessage?.also { error ->
+            Column(
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement =
+                    Arrangement.spacedBy(6.dp, alignment = Alignment.CenterVertically),
+                modifier = Modifier.fillMaxWidth().height(200.dp)) {
+                  Icon(
+                      painter = painterResource(id = R.drawable.warning),
+                      modifier = Modifier.size(48.dp),
+                      contentDescription = null,
+                      tint = Color.Red)
+                  Text(
+                      stringResource(id = R.string.pingFailed),
+                      fontSize = MaterialTheme.typography.titleMedium.fontSize,
+                      fontFamily = MaterialTheme.typography.titleMedium.fontFamily,
+                      fontWeight = FontWeight.Bold,
+                      textAlign = TextAlign.Center,
+                      color = Color.Red)
+                  Text(
+                      error,
```

Comment:
The error that comes back here isn't really human readable when you ping an offline device (and I will assume for other localAPI errors).  We should render something canned "Unable to reach device" or just elide the detailed error altogether.

go.Universe$proxyError<blah> or something.

Reply:
Fixed this, errors now get remapped to either of two strings (depending on whether "timeout" is returned within the error string)